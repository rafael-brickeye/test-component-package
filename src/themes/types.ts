export type FontSizeTypes = 'xs' | 'sm' | 'md' | 'lg' | 'xl' | '2xl' | '3xl' | '4xl' | '5xl';
export type SpacingTypes =
  | 's-2'
  | 's-4'
  | 's-8'
  | 's-12'
  | 's-16'
  | 's-20'
  | 's-24'
  | 's-28'
  | 's-32'
  | 's-36'
  | 's-40';
