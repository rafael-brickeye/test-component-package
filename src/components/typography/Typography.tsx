import React from 'react';
import {Text, TextProps, TextStyle} from 'react-native';
import useStyles from './Typography.styles';

type TextDefaultProps = TextProps & TextStyle;

export type TypographyPropTypes = {
  truncateEnd?: boolean;
} & TextDefaultProps;

const Typography: React.FC<TypographyPropTypes> = ({children, truncateEnd = false, style: styleProps, ...props}) => {
  const style = useStyles({...props});

  return (
    <Text
      numberOfLines={truncateEnd ? 1 : 0}
      style={{...style.typographyContainer, ...(styleProps as TextStyle)}}
      {...props}>
      {children}
    </Text>
  );
};

export {Typography};
