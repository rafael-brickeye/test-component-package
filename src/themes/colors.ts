// All values used here come from Figma file design styling

const background = {
  white: '#ffffff',
};

const surface = {
  grey: '#C5CCD6',
};

const text = {
  black: '#000000DE',
  grey: '#71727A',
  white: '#ffffff',
};

const brand = {
  protect: {
    primary: {
      main: '#0048BB',
    },
  },
  lumicon: {
    primary: {
      light: '#FF9F34',
      main: '#FF8C4D',
      dark: '#FF5968',
    },
    secondary: {
      light: '#F46167',
      main: '#F05E40',
      dark: '#9E9E9E',
    },
    info: {
      light: '#64B6F7',
      main: '#2196F3',
      dark: '#0B79D0',
    },
    error: {
      light: '#F88078',
      main: '#F44336',
      dark: '#E31B0C',
    },
    warning: {
      light: '#FFB547',
      main: '#FF9800',
      dark: '#C77700',
    },
    success: {
      light: '#7BC67E',
      main: '#4CAF50',
      dark: '#3B873E',
    },
  },
};

const shadow = {
  default: {
    shadowColor: '#000',
    shadowOffset: {
      width: 0,
      height: 1,
    },
    shadowOpacity: 0.18,
    shadowRadius: 1.0,
    elevation: 1,
  },
  wide: {
    shadowColor: '#000',
    shadowOffset: {
      width: 0,
      height: 2,
    },
    shadowOpacity: 0.23,
    shadowRadius: 2.62,
    elevation: 4,
  },
};

export const color = {text, brand, surface, background, shadow};
