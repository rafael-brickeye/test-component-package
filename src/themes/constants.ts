import {TextStyle, ViewStyle} from 'react-native';

export const defaultViewStyles: Record<keyof Omit<ViewStyle, 'testID'>, string> = {
  alignContent: 'alignContent',
  alignItems: 'alignItems',
  alignSelf: 'alignSelf',
  aspectRatio: 'aspectRatio',
  backfaceVisibility: 'backfaceVisibility',
  backgroundColor: 'backgroundColor',
  borderBottomColor: 'borderBottomColor',
  borderBottomEndRadius: 'borderBottomEndRadius',
  borderBottomLeftRadius: 'borderBottomLeftRadius',
  borderBottomRightRadius: 'borderBottomRightRadius',
  borderBottomStartRadius: 'borderBottomStartRadius',
  borderBottomWidth: 'borderBottomWidth',
  borderColor: 'borderColor',
  borderEndColor: 'borderEndColor',
  borderEndWidth: 'borderEndWidth',
  borderLeftColor: 'borderLeftColor',
  borderLeftWidth: 'borderLeftWidth',
  borderRadius: 'borderRadius',
  borderRightColor: 'borderRightColor',
  borderRightWidth: 'borderRightWidth',
  borderStartColor: 'borderStartColor',
  borderStartWidth: 'borderStartWidth',
  borderStyle: 'borderStyle',
  borderTopColor: 'borderTopColor',
  borderTopEndRadius: 'borderTopEndRadius',
  borderTopLeftRadius: 'borderTopLeftRadius',
  borderTopRightRadius: 'borderTopRightRadius',
  borderTopStartRadius: 'borderTopStartRadius',
  borderTopWidth: 'borderTopWidth',
  borderWidth: 'borderWidth',
  bottom: 'bottom',
  direction: 'direction',
  display: 'display',
  elevation: 'elevation',
  end: 'end',
  flex: 'flex',
  flexBasis: 'flexBasis',
  flexDirection: 'flexDirection',
  flexGrow: 'flexGrow',
  flexShrink: 'flexShrink',
  flexWrap: 'flexWrap',
  height: 'height',
  justifyContent: 'justifyContent',
  left: 'left',
  margin: 'margin',
  marginBottom: 'marginBottom',
  marginEnd: 'marginEnd',
  marginHorizontal: 'marginHorizontal',
  marginLeft: 'marginLeft',
  marginRight: 'marginRight',
  marginStart: 'marginStart',
  marginTop: 'marginTop',
  marginVertical: 'marginVertical',
  maxHeight: 'maxHeight',
  maxWidth: 'maxWidth',
  minHeight: 'minHeight',
  minWidth: 'minWidth',
  opacity: 'opacity',
  overflow: 'overflow',
  padding: 'padding',
  paddingBottom: 'paddingBottom',
  paddingEnd: 'paddingEnd',
  paddingHorizontal: 'paddingHorizontal',
  paddingLeft: 'paddingLeft',
  paddingRight: 'paddingRight',
  paddingStart: 'paddingStart',
  paddingTop: 'paddingTop',
  paddingVertical: 'paddingVertical',
  position: 'position',
  right: 'right',
  rotation: 'rotation',
  scaleX: 'scaleX',
  scaleY: 'scaleY',
  shadowColor: 'shadowColor',
  shadowOffset: 'shadowOffset',
  shadowOpacity: 'shadowOpacity',
  shadowRadius: 'shadowRadius',
  start: 'start',
  top: 'top',
  transform: 'transform',
  transformMatrix: 'transformMatrix',
  translateX: 'translateX',
  translateY: 'translateY',
  width: 'width',
  zIndex: 'zIndex',
};

export const defaultTextStyles: Record<keyof Omit<TextStyle, 'testID'>, string> = {
  alignContent: 'alignContent',
  alignItems: 'alignItems',
  alignSelf: 'alignSelf',
  aspectRatio: 'aspectRatio',
  backfaceVisibility: 'backfaceVisibility',
  backgroundColor: 'backgroundColor',
  borderBottomColor: 'borderBottomColor',
  borderBottomEndRadius: 'borderBottomEndRadius',
  borderBottomLeftRadius: 'borderBottomLeftRadius',
  borderBottomRightRadius: 'borderBottomRightRadius',
  borderBottomStartRadius: 'borderBottomStartRadius',
  borderBottomWidth: 'borderBottomWidth',
  borderColor: 'borderColor',
  borderEndColor: 'borderEndColor',
  borderEndWidth: 'borderEndWidth',
  borderLeftColor: 'borderLeftColor',
  borderLeftWidth: 'borderLeftWidth',
  borderRadius: 'borderRadius',
  borderRightColor: 'borderRightColor',
  borderRightWidth: 'borderRightWidth',
  borderStartColor: 'borderStartColor',
  borderStartWidth: 'borderStartWidth',
  borderStyle: 'borderStyle',
  borderTopColor: 'borderTopColor',
  borderTopEndRadius: 'borderTopEndRadius',
  borderTopLeftRadius: 'borderTopLeftRadius',
  borderTopRightRadius: 'borderTopRightRadius',
  borderTopStartRadius: 'borderTopStartRadius',
  borderTopWidth: 'borderTopWidth',
  borderWidth: 'borderWidth',
  bottom: 'bottom',
  color: 'color',
  direction: 'direction',
  display: 'display',
  elevation: 'elevation',
  end: 'end',
  flex: 'flex',
  flexBasis: 'flexBasis',
  flexDirection: 'flexDirection',
  flexGrow: 'flexGrow',
  flexShrink: 'flexShrink',
  flexWrap: 'flexWrap',
  fontFamily: 'fontFamily',
  fontSize: 'fontSize',
  fontStyle: 'fontStyle',
  fontVariant: 'fontVariant',
  fontWeight: 'fontWeight',
  height: 'height',
  includeFontPadding: 'includeFontPadding',
  justifyContent: 'justifyContent',
  left: 'left',
  letterSpacing: 'letterSpacing',
  lineHeight: 'lineHeight',
  margin: 'margin',
  marginBottom: 'marginBottom',
  marginEnd: 'marginEnd',
  marginHorizontal: 'marginHorizontal',
  marginLeft: 'marginLeft',
  marginRight: 'marginRight',
  marginStart: 'marginStart',
  marginTop: 'marginTop',
  marginVertical: 'marginVertical',
  maxHeight: 'maxHeight',
  maxWidth: 'maxWidth',
  minHeight: 'minHeight',
  minWidth: 'minWidth',
  opacity: 'opacity',
  overflow: 'overflow',
  padding: 'padding',
  paddingBottom: 'paddingBottom',
  paddingEnd: 'paddingEnd',
  paddingHorizontal: 'paddingHorizontal',
  paddingLeft: 'paddingLeft',
  paddingRight: 'paddingRight',
  paddingStart: 'paddingStart',
  paddingTop: 'paddingTop',
  paddingVertical: 'paddingVertical',
  position: 'position',
  right: 'right',
  rotation: 'rotation',
  scaleX: 'scaleX',
  scaleY: 'scaleY',
  shadowColor: 'shadowColor',
  shadowOffset: 'shadowOffset',
  shadowOpacity: 'shadowOpacity',
  shadowRadius: 'shadowRadius',
  start: 'start',
  textAlign: 'textAlign',
  textAlignVertical: 'textAlignVertical',
  textDecorationColor: 'textDecorationColor',
  textDecorationLine: 'textDecorationLine',
  textDecorationStyle: 'textDecorationStyle',
  textShadowColor: 'textShadowColor',
  textShadowOffset: 'textShadowOffset',
  textShadowRadius: 'textShadowRadius',
  textTransform: 'textTransform',
  top: 'top',
  transform: 'transform',
  transformMatrix: 'transformMatrix',
  translateX: 'translateX',
  translateY: 'translateY',
  width: 'width',
  writingDirection: 'writingDirection',
  zIndex: 'zIndex',
};
