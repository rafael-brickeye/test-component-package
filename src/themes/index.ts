export * from './colors';
export * from './sizing';
export * from './spacing';
export * from './utils';
export * from './types';
export * from './constants';
