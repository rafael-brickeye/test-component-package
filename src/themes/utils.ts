import {TextStyle, ViewStyle} from 'react-native';
import {defaultTextStyles, defaultViewStyles} from './constants';

type DefaultStyles = ViewStyle | TextStyle;

const MULTIPLE_OF_TWO = 2;
const MULTIPLE_OF_FOUR = 4;

export const generateSpacing = (value: number) => MULTIPLE_OF_FOUR * value;

export const generateSize = (value: number) => MULTIPLE_OF_TWO * value;

export const getFontSize = (value: number) => generateSize(value * 1);

export const getDefaultStylesFromProps = (props: DefaultStyles) => {
  const newProps: Record<string, any> = {};

  for (const validProp in props) {
    const propValue = props[validProp as keyof DefaultStyles];
    const isValidProp = validProp in defaultViewStyles || validProp in defaultTextStyles;

    if (isValidProp) {
      newProps[validProp] = propValue;
    }
  }

  return newProps;
};
