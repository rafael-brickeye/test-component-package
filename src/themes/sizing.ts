import {FontSizeTypes} from './types';
import {getFontSize} from './utils';

const font: Record<FontSizeTypes, number> = {
  xs: getFontSize(4),
  sm: getFontSize(5),
  md: getFontSize(6),
  lg: getFontSize(7),
  xl: getFontSize(8),
  '2xl': getFontSize(9),
  '3xl': getFontSize(10),
  '4xl': getFontSize(11),
  '5xl': getFontSize(12),
};

export const sizing = {font};
