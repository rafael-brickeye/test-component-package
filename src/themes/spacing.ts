import {SpacingTypes} from './types';
import {generateSpacing} from './utils';

const spacing: Record<SpacingTypes, number> = {
  's-2': generateSpacing(0.5),
  's-4': generateSpacing(1),
  's-8': generateSpacing(2),
  's-12': generateSpacing(3),
  's-16': generateSpacing(4),
  's-20': generateSpacing(5),
  's-24': generateSpacing(6),
  's-28': generateSpacing(7),
  's-32': generateSpacing(8),
  's-36': generateSpacing(9),
  's-40': generateSpacing(10),
};

export {spacing};
