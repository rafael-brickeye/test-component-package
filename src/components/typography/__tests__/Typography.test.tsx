import React from 'react';
import {View} from 'react-native';
import {cleanup, render} from '@testing-library/react-native';
import '@testing-library/jest-native';
import {Typography} from '../Typography';
import {sizing} from '../../../themes';

afterEach(cleanup);

const testText = 'Hello World';
const longTestText = testText.repeat(5);
const testId = 'typography-component';

describe('Base Typography Component', () => {
  it('should match the existing snapshot', () => {
    const testSnapshot = render(<Typography />).toJSON();

    expect(testSnapshot).toMatchSnapshot();
  });

  it('should be able to render child component', () => {
    const {getByTestId} = render(<Typography testID={testId}>{testText}</Typography>);

    expect(getByTestId(testId)).toBeVisible();
    expect(getByTestId(testId)).toHaveTextContent(testText);
  });

  it("should truncate text when it's about to overflow the max content of the parent component and truncateEnd prop is TRUE", () => {
    const {children} = render(
      <View style={{width: 5}}>
        <Typography testID={testId} truncateEnd>
          {longTestText}
        </Typography>
      </View>,
    ).toJSON() as any;

    const typographyProps = children[0]?.props;

    expect(typographyProps).toHaveProperty('numberOfLines', 1);
  });

  it('should be able to use preset styling and override default styling', () => {
    const {getByTestId} = render(
      <Typography testID={testId} fontWeight={'bold'} fontSize={sizing.font['4xl']}>
        {testText}
      </Typography>,
    );

    expect(getByTestId(testId)).not.toHaveStyle({fontSize: sizing.font.md, fontWeight: '500'});
    expect(getByTestId(testId)).toHaveStyle({fontSize: 22, fontWeight: 'bold'});
  });
});
