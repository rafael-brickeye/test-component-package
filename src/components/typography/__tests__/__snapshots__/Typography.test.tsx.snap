// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Base Typography Component should match the existing snapshot 1`] = `
<Text
  numberOfLines={0}
  style={
    Object {
      "color": "#000000DE",
      "fontSize": 12,
      "fontWeight": "normal",
    }
  }
/>
`;
